src/main/java/org/folio/bulkops/service/QueryService.java

```diff
@@ -1,94 +1,318 @@
+import org.folio.bulkops.domain.bean.ExtendedHoldingsRecord;
+import org.folio.bulkops.domain.bean.ExtendedInstance;
+import org.folio.bulkops.domain.bean.ExtendedItem;
+import org.folio.bulkops.domain.bean.HoldingsRecord;
+import org.folio.bulkops.domain.bean.Instance;
+import org.folio.bulkops.domain.bean.Item;
+import org.folio.bulkops.domain.bean.StateType;
+import org.folio.bulkops.domain.bean.User;
+import org.folio.bulkops.domain.converter.JsonToMarcConverter;
 import org.folio.bulkops.domain.entity.BulkOperation;
+import org.folio.bulkops.domain.entity.BulkOperationExecutionContent;
+import org.folio.bulkops.exception.UploadFromQueryException;
+import org.folio.bulkops.processor.permissions.check.PermissionsValidator;
 import org.folio.bulkops.repository.BulkOperationRepository;
+import org.folio.bulkops.util.BulkOperationsEntityCsvWriter;
+import org.folio.bulkops.util.CSVHelper;
+import org.folio.querytool.domain.dto.QueryDetails;
 import org.folio.querytool.domain.dto.SubmitQuery;
+import org.folio.spring.FolioExecutionContext;
 import org.springframework.stereotype.Service;
 
-import java.io.ByteArrayInputStream;
-import java.time.LocalDateTime;
-import java.util.UUID;
-import java.util.concurrent.ExecutorService;
-import java.util.concurrent.Executors;
-import java.util.stream.Collectors;
-
 @Service
 @Log4j2
 @RequiredArgsConstructor
 public class QueryService {
   public static final String QUERY_FILENAME_TEMPLATE = "%1$s/Query-%1$s.csv";
 
-  private final QueryClient queryClient;
   private final BulkOperationRepository bulkOperationRepository;
-  private final RemoteFileSystemClient remoteFileSystemClient;
   private final ErrorService errorService;
 
+  private final FolioExecutionContext folioExecutionContext;
+
+  private final ObjectMapper objectMapper;
+
+  private final JsonToMarcConverter jsonToMarcConverter;
+
+  private final PermissionsValidator permissionsValidator;
+
+  private final RemoteFileSystemClient remoteFileSystemClient;
+  private final SrsClient srsClient;
+  private final QueryClient queryClient;
+
   private final ExecutorService executor = Executors.newCachedThreadPool();
 
-  public UUID executeQuery(SubmitQuery submitQuery) {
+  public BulkOperation retrieveRecordsAndCheckQueryExecutionStatus(BulkOperation bulkOperation) {
+    executor.execute(getRunnableWithCurrentFolioContext(() -> {
+      var queryId = executeQuery(new SubmitQuery(bulkOperation.getFqlQuery(), bulkOperation.getEntityTypeId()));
+      var queryResult = getResult(queryId);
+      switch (queryResult.getStatus()) {
+          case IN_PROGRESS -> log.info("Retrieving records by FKM for operation {} in progress...", bulkOperation.getId());
+          case SUCCESS -> {
+          if (queryResult.getTotalRecords() == 0) {
+            failBulkOperation(bulkOperation, "No records found for the query");
+          } else {
+            saveIdentifiersAndStartBulkOperation(bulkOperation, queryResult);
```

Comment:
Only in this if-branch is the content required. Here, a second call to /query/{{queryId}}?includeResults=true should be made.